packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,428 @@
+    state: recommendationListState,
+    subscribe: functionMocks.subscribe,
+  })),
+  buildResultTemplatesManager: jest.fn(() => ({
+    registerTemplates: functionMocks.registerTemplates,
+  })),
+  loadFieldActions: jest.fn(() => ({
+    registerFieldsToInclude: functionMocks.registerFieldsToInclude,
+  })),
+  subscribe: jest.fn((callback) => {
+    callback();
+    return functionMocks.unsubscribe;
+  }),
+  unsubscribe: jest.fn(() => {}),
+  dispatch: jest.fn(() => {}),
+  registerTemplates: jest.fn(() => {
+    return {
+      content: jest.fn(),
+      conditions: [],
+    };
+  }),
+  registerFieldsToInclude: jest.fn((fieldsToInclude) => {
+    return {
+      fieldsToInclude,
+    };
+  }),
+};
+
+const exampleEngine = {
+  id: 'exampleEngineId',
+  dispatch: functionMocks.dispatch,
+};
+const defaultFieldsToInclude = 'field1,field2';
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  recommendation: 'recommendation',
+  numberOfRecommendations: 10,
+  label: 'Top documents for you',
+  variant: 'grid',
+  recommendationsPerRow: 3,
+  fieldsToInclude: defaultFieldsToInclude,
+};
+
+const selectors = {
+  initializationError: 'c-quantic-component-error',
+  gridContainer: '[data-testid="recommendations__grid"]',
+  carouselContainer: '[data-testid="recommendations__carousel"]',
+  recommendationListItem: '[data-testid="recommendations__item"]',
+  placeholder: 'c-quantic-placeholder',
+  quanticHeading: 'c-quantic-heading',
+};
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticRecommendationList,
+  'c-quantic-recommendation-list',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildRecommendationList: functionMocks.buildRecommendationList,
+      buildResultTemplatesManager: functionMocks.buildResultTemplatesManager,
+      loadFieldActions: functionMocks.loadFieldActions,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticRecommendationList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticRecommendationList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-recommendation-list', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+    console.error = jest.fn();
+  });
+
+  afterEach(() => {
+    cleanup();
+    isInitialized = false;
+    recommendationListState = initialRecommendationListState;
+  });
+
+  describe('when the component is loading', () => {
+    it('should display the placeholder component when loading', async () => {
```

Comment:
it should not display the placeholder when there is an error.

You're missing that test.

Reply:
The placeholder also has a different behavior depending on if the variant is grid or caroussel.